extensions/s3/src/main/java/io/deephaven/extensions/s3/ProfileCredentials.java

```diff
@@ -0,0 +1,46 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.extensions.s3;
+
+import io.deephaven.annotations.CopyableStyle;
+import org.immutables.value.Value;
+import software.amazon.awssdk.auth.credentials.ProfileCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.profiles.ProfileFile;
+
+import java.util.Optional;
+
+/**
+ * A set of credentials that are sourced from a profile file. This class is not public since it is not intended to be
+ * used directly by users. Instead, users should use the {@link S3Instructions} class to provide config and credentials
+ * files.
+ */
+@Value.Immutable
+@CopyableStyle
```

Comment:
I don't see the use of copy here; I think `@BuildableStyle` is more appropriate.